### PR TITLE
doc/architecture: edit "OSD Membership and Status"

### PR DESCRIPTION
**Pull request by zdover23**
Edit "OSD Membership and Status" in the "Smart Daemons Enable Hyperscale" section of doc/architecture.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -403,20 +403,22 @@ the greater cluster provides several benefits:
...
-   Daemon cannot notify the Ceph Monitor that it is ``down``. The OSDs
-   periodically send messages to the Ceph Monitor (``MPGStats`` pre-luminous,
-   and a new ``MOSDBeacon`` in luminous).  If the Ceph Monitor doesn't see that
-   message after a configurable period of time then it marks the OSD down.
-   This mechanism is a failsafe, however. Normally, Ceph OSD Daemons will
-   determine if a neighboring OSD is down and report it to the Ceph Monitors.
-   This assures that Ceph Monitors are lightweight processes.  See `Monitoring
-   OSDs`_ and `Heartbeats`_ for additional details.
+#. **OSD Membership and Status**: When Ceph OSD Daemons join a cluster, they
+   report on their status. At the lowest level, the Ceph OSD Daemon status is
```
Comment (anthonyeleven) [inline doc/architecture.rst:407]: s/on //

File: doc/architecture.rst [hunk 2]
```diff
@@ -403,20 +403,22 @@ the greater cluster provides several benefits:
...
+   ``up`` or ``down``: this reflects whether the Ceph OSD daemon is running and
+   able to service Ceph Client requests. If a Ceph OSD Daemon is ``down`` and
+   ``in`` the Ceph Storage Cluster, this status may indicate the failure of the
+   Ceph OSD Daemon. If a Ceph OSD Daemon is not running (for example, because
+   it has crashed), the Ceph OSD Daemon cannot notify the Ceph Monitor that it
+   is ``down``. The OSDs periodically send messages to the Ceph Monitor (in
```
Comment (anthonyeleven) [inline doc/architecture.rst:413]: Change parenthetical to a sentence

File: doc/architecture.rst [hunk 3]
```diff
@@ -403,20 +403,22 @@ the greater cluster provides several benefits:
...
+   releases prior to Luminous, this was done by means of ``MPGStats``, and
+   beginning with the Luminous release, this has been done with
+   ``MOSDBeacon``). If the Ceph Monitor doesn't receive such a message after a
```
Comment (anthonyeleven) [inline doc/architecture.rst:416]: Plural

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:407]: That is better.

File: doc/architecture.rst (hunk 2 above)
Comment (zdover23) [reply doc/architecture.rst:413]: Oh, very well.

File: doc/architecture.rst (hunk 3 above)
Comment (zdover23) [reply doc/architecture.rst]: I've rewritten the sentence, making "Monitor" into "Monitors".

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53727 - Reef backport
https://github.com/ceph/ceph/pull/53728 - Quincy backport

**Pull request merged**